addons/website/static/src/snippets/s_social_media/options.js

```diff
@@ -129,10 +131,27 @@ options.registry.SocialMedia = options.Class.extend({
             return this._super(methodName, params);
         }
         const listEntries = [];
+        for (const socialMedia of Object.keys(this.dbSocialValues)) {
+            const mediaName = socialMedia.split('social_')[1];
+            const dbEl = this.$target[0].querySelector(`a[href="/website/social/${mediaName}"]`);
+            if (!dbEl) {
+                // If a social media exists in DB but not in target, create it.
+                const anchorEl = this.$target[0].querySelector(':scope > a').cloneNode(true);
+                anchorEl.href = `/website/social/${mediaName}`;
+                this._removeSocialMediaClasses(anchorEl);
+                anchorEl.classList.add(`s_social_media_${mediaName}`, 'd-none');
+                anchorEl.querySelector('i').classList.add(`fa-${mediaName}`);
+                this.$target[0].appendChild(anchorEl);
+                this.links.push({id: generateHTMLId(), el: anchorEl});
```

Comment:
I doubt this part should be here.
This is called everytime:
- You focus the snippet
- You change an option of the social media
While it works I guess, that seems too much.

On the other hand, this is not called when drag & dropping the snippet.

There should definitely have better place to do that I guess, @qsm-odoo ?

Reply:
@rdeodoo @Guillaume-gdi Well of course... `_computeWidgetState` is supposed to return the value a widget should have (for a specific method) according to the current DOM. -> It should not update the DOM. Difficult for me to say where that should be because I don't understand the point of the code.
> // If a social media exists in DB but not in target, create it.

No? Just display it as an option in the right panel and if the user interacts with it, then it should create the related DOM? Again, not sure to understand the code, I may be mistaken.

Reply:
Ok I understand that I can't modify the DOM in this function, so with quentin's solution, the link creation part is only done when you toogle in the we-list, good for me :+1: . 
On the other hand for the part of hiding the links that are in the DOM but not in DB I don't see a good/simple way to do that since I can't touch the DOM in willStart and here. Currently we have the weird behavior that the link will be hidden only after the user has had a first interaction with the we-list.
So my question is: Which function can modify the DOM and is called before the first interaction of the user with the we-list? Or is there another way to fix this?

Initial comment : https://github.com/odoo/odoo/pull/80217#discussion_r786675091

Reply:
onBuilt